### PR TITLE
New BeforeAfter image slider

### PR DESCRIPTION
**Pull request by AnnMarieW**
Here is the new BeforeAfter image slider based on https://github.com/sneas/img-comparison-slider

`npm install --save @img-comparison-slider/react`

What other html props do you think should be added to the `img` components?

### CODE REVIEW

File: src/lib/components/BeforeAfter.react.js [hunk 1]
```diff
@@ -1,55 +1,153 @@
...
 
-    // Dash props.
+    /**
+     * className of after image. Often used with CSS to style elements with common properties.
+     */
+    afterClassName: PropTypes.string,
 
     /**
-     * The ID used to identify this component in Dash callbacks
+     * **DEPRECATED** Use `value` instead.
```
Comment (emilhe) [inline src/lib/components/BeforeAfter.react.js:132]: Now that we are moving to a new component, I think we should just remove the property. As we discussed the changes (e.g. switching before/after order) are breaking already, so I'll just include that in the release note.

File: src/lib/components/BeforeAfter.react.js [hunk 2]
```diff
@@ -1,55 +1,153 @@
...
+    /**
+     * Enable/disable slider position control with the keyboard
+     */
+    keyboard: PropTypes.oneOf(['enabled', 'disabled']),
 
-    afterClassName: PropTypes.string,
+    /**
+     * Defines CSS styles of the before image.
+     */
+    before_style: PropTypes.object,
```
Comment (emilhe) [inline src/lib/components/BeforeAfter.react.js:114]: Should we name the `before_style` prop as `beforeStyle` (and after vice versa) for consistency?

**Comment by emilhe** [comment]
An alternative to the current design using `beforeClassName`, `beforeStyle` etc. could be to use `beforeProps` as before, and then just pass these props to the first `img` element (and the second with after vice versa). That would be more flexible in the sense that you would support all html properties out-of-the-box, but probably a little less user friendly for new Dash user. What do you think, @AnnMarieW ?

File: src/lib/components/BeforeAfter.react.js (hunk 2 above)
Comment (AnnMarieW) [reply src/lib/components/BeforeAfter.react.js]: I removed the `before_style` and `after_style` props  and went with your suggestion below of using `beforeProps` and `afterProps`  That's a good idea and makes all the `html.Img` props available without being super verbose.  

I think including an example in the docs of how to use these props will make it beginner friendly.

File: src/lib/components/BeforeAfter.react.js [hunk 3]
```diff
@@ -1,55 +1,137 @@
...
+                direction={direction}
+                keyboard={keyboard}
+            >
+                <img
+                    slot="first"
+                    width={width}
+                    height={height}
+                    src={before}
+                    {...beforeProps}
+                    className={beforeClassName}
```
Comment (emilhe) [inline src/lib/components/BeforeAfter.react.js:39]: Shouldn't we include the class in the "before/after" props? Or is there a reson why you left it out?

Comment (AnnMarieW) [reply src/lib/components/BeforeAfter.react.js:39]: I'm not sure what you mean.  This matches the props of the original component.

Comment (emilhe) [reply src/lib/components/BeforeAfter.react.js:39]: I was just thinking if we should include the "className" in the "beforeProps" (and "afterProps")? :)

Comment (AnnMarieW) [reply src/lib/components/BeforeAfter.react.js:39]: OK, I get it now.  Your call.  I was trying to match the current props so fewer would be  deprecated, but including className in `beforeProps` and/or `afterProps` makes sense too.

Comment (emilhe) [reply src/lib/components/BeforeAfter.react.js]: A minor thing - the static code analysis picked up an unused variable,

image

Comment (AnnMarieW) [reply src/lib/components/BeforeAfter.react.js]: oh darn, I thought I took that out.  I'll fix it

Comment (AnnMarieW) [reply src/lib/components/BeforeAfter.react.js]: Actually - I already did fix that - it's not in the current version of the file

**Pull request merged**